### PR TITLE
Virginia military benefit subtraction

### LINKED ISSUE #2240: Virginia Military Benefit Subtraction
age 55 or older.
Year 2022: $10,000 max
Year 2023: $20,000 max
Year 2024: $30,000 max
Year 2025 and after: $40,000 max
No other credit, exemption, subtraction, or deduction is claimed for the same income				
[https://www.tax.virginia.gov/sites/default/files/vatax-pdf/2022-760-instructions.pdf#page=28] 60
§ 58.1-322.02. (18.b)

### PR DESCRIPTION
**Pull request by oliverh233**
Fixes #2240

### CODE REVIEW

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/military_benefit/amount.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+description: Virginia provides a military benefit subtraction up to this amount for military personnel receiving military benefit and 55 or older.
+
+values:
+  2022-01-01: 10_000
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/military_benefit/amount.yaml:4]: add values for the following years as it is in the source,

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/military_benefit/age_threshold.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+description: Virginia provides a military benefit subtraction for military personnel of this age or older.
+
+values:
+  2022-01-01: 55
+
+metadata:
+  unit: year
+  label: Age threshold for Virginia military benefit subtraction
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/military_benefit/age_threshold.yaml:8]:
```suggestion
  label: Virginia military benefit subtraction age threshold
```

File: policyengine_us/variables/gov/states/va/tax/income/subtractions/va_military_benefit_subtraction.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        person = tax_unit.members
+        p = parameters(
+            period
+        ).gov.states.va.tax.income.subtractions.military_benefit
+
+        subtractable_military_benefit = min_(
+            person("military_benefit", period), p.amount
+        )
+
+        age = person("age", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/subtractions/va_military_benefit_subtraction.py:25]: we can use the `age_head` tax_unit variable for lines 25-27

File: policyengine_us/variables/gov/states/va/tax/income/subtractions/va_military_benefit_subtraction.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        head = person("is_tax_unit_head", period)
+        head_age = tax_unit.max(age * head)
+
+        spouse = person("is_tax_unit_spouse", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/subtractions/va_military_benefit_subtraction.py:29]: we can use the `age_spouse` tax_unit variable for lines 29-30

File: policyengine_us/variables/household/income/person/misc/military_benefit.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from policyengine_us.model_api import *
+
+
+class military_benefit(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/household/income/person/misc/military_benefit.py:4]:
```suggestion
class military_benefits(Variable):
```

File: policyengine_us/variables/household/income/person/misc/military_benefit.py [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+    value_type = float
+    entity = Person
+    label = "Military benefit"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/household/income/person/misc/military_benefit.py:7]:
```suggestion
    label = "Military benefits"
```

File: policyengine_us/variables/household/income/person/misc/military_benefit.py (hunk 5 above)
Comment (PavelMakarchuk) [reply policyengine_us/variables/household/income/person/misc/military_benefit.py:4]: rename to `military_benefits.py`

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/military_benefit/amount.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+description: Virginia provides a military benefit subtraction up to this amount for military personnel receiving military benefit and 55 or older.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/military_benefit/amount.yaml:1]:
```suggestion
description: Virginia provides a military benefit subtraction up to this amount.
```

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/military_benefit/amount.yaml [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+values:
+  2022-01-01: 10_000
+  2023-01-01: 20_000
+  2024-01-01: 30_000
+  2025-01-01: 40_000
+
+metadata:
+  period: year
+  unit: currency-USD
+  label: Virginia military benefit subtraction
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/military_benefit/amount.yaml:12]:
```suggestion
  label: Virginia military benefit subtraction amount
```

File: policyengine_us/variables/gov/states/va/tax/income/subtractions/va_military_benefit_subtraction.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    )
+
+    def formula(tax_unit, period, parameters):
+        person = tax_unit.members
+        p = parameters(
+            period
+        ).gov.states.va.tax.income.subtractions.military_benefit
+
+        subtractable_military_benefit = min_(
+            person("military_benefits", period), p.amount
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/subtractions/va_military_benefit_subtraction.py:22]:
```suggestion
            person("military_retirement_pay", period), p.amount
```

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/va/tax/income/subtractions/va_military_benefit_subtraction.py:22]: we can use this variable instead as it already exists

File: policyengine_us/variables/household/income/person/misc/military_benefits.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/household/income/person/misc/military_benefits.py:1]: delete

**Review by MaxGhenis — APPROVED** [review]
Approved conditional on Pavel's suggestions

**Pull request merged**